src/core/kmod-setup.c

```diff
@@ -128,6 +128,48 @@ static bool has_virtio_vsock(void) {
         return r > 0;
 }
 
+static bool has_virtiofs(void) {
+        int r;
+
+        /* Directory traversal might be slow, hence let's do a cheap check first if it's even worth it */
+        if (detect_vm() == VIRTUALIZATION_NONE)
+                return false;
+
+        r = recurse_dir_at(
+                        AT_FDCWD,
+                        "/sys/devices/pci0000:00",
+                        /* statx_mask= */ 0,
+                        /* n_depth_max= */ 3,
+                        RECURSE_DIR_ENSURE_TYPE,
+                        match_modalias_recurse_dir_cb,
+                        STRV_MAKE("virtio:d0000001Av"));
```

Comment:
this becomes a bit too repetitive... can you add `has_virtio_feature(const char *modalias)` please that wraps the recurse_dir_at() and the detect_vm() call? And then just make the individual tests:

```c
static bool has_virtiofs(void) {
        return has_virtio_feature("virtio:d0000001Av");
}
``